scripts/utils/deploy-store.js

```diff
@@ -0,0 +1,49 @@
+const { writeFile } = require('./file');
+const path = require('path');
+
+const getDeployFilePath = (network, contractName) => {
+  return path.resolve(
+    __dirname,
+    `../../deployments/${network}/${contractName}.json`
+  );
+};
+
+module.exports.deployStore = async (network, contractName, contract) => {
+  const filePath = getDeployFilePath(network, contractName);
+
+  const buildArtifact = await hre.artifacts.readArtifact(contractName);
+  const {
+    id: buildId,
+    input: solcInput,
+    output: solcOutput,
+  } = await hre.artifacts.getBuildInfo(
+    `${buildArtifact.sourceName}:${buildArtifact.contractName}`
+  );
+
+  const contractOutput =
+    solcOutput.contracts[buildArtifact.sourceName][contractName];
+
+  const data = {
+    timestamp: new Date().toLocaleString('en-US'),
+    address: contract.address,
+    transactionHash: contract.deployTransaction.hash,
+    args: contract.deployTransaction.args,
+    gasPrice: contract.deployTransaction.gasPrice.toNumber(),
+    abi: contractOutput.abi,
+    bytecode: buildArtifact.bytecode,
+    metadata: contractOutput.metadata,
+    storageLayout: contractOutput.storageLayout,
+  };
+
+  try {
```

Comment:
Does the above code need `try`?

Reply:
It is not required, but with it we are identifying that the error happened during the file write, it may help debugging when the error happens when this is script is called by others.